react/src/component/PerfTestDashboard.jsx

```diff
@@ -8,83 +8,113 @@ import axios from "@/axios";
-    { value: 'networkingCost', label: 'networkingCost', color: '#800000' },
-    { value: 'appStateCost', label: 'appStateCost', color: '#FFCC00' },
-    { value: 'locationCost', label: 'locationCost', color: '#808000' },
-    { value: 'thermalStateCost', label: 'thermalStateCost', color: '#4B0080' },
-    { value: 'totalOverhead', label: 'totalOverhead', color: '#8884d8' },
-    { value: 'cpuOverhead', label: 'cpuOverhead', color: '#8B008B' },
-    { value: 'gpuOverhead', label: 'gpuOverhead', color: '#EE7600' },
-    { value: 'networkingOverhead', label: 'networkingOverhead', color: '#CD5C5C' },
-    { value: 'appStateOverhead', label: 'appStateOverhead', color: '#BC8F8F' },
-    { value: 'locationOverhead', label: 'locationOverhead', color: '#8B8B7A' },
-    { value: 'thermalStateOverhead', label: 'thermalStateOverhead', color: '#006400' }
+    { value: 'totalCost', label: 'totalCost', color: COLORS[0] },
+    { value: 'cpuCost', label: 'cpuCost', color: COLORS[1] },
+    { value: 'gpuCost', label: 'gpuCost', color: COLORS[2] },
+    { value: 'networkingCost', label: 'networkingCost', color: COLORS[3] },
+    { value: 'appStateCost', label: 'appStateCost', color: COLORS[4] },
+    { value: 'locationCost', label: 'locationCost', color: COLORS[5] },
+    { value: 'thermalStateCost', label: 'thermalStateCost', color: COLORS[6] },
+    { value: 'totalOverhead', label: 'totalOverhead', color: COLORS[7] },
+    { value: 'cpuOverhead', label: 'cpuOverhead', color: COLORS[8] },
+    { value: 'gpuOverhead', label: 'gpuOverhead', color: COLORS[9] },
+    { value: 'networkingOverhead', label: 'networkingOverhead', color: COLORS[10] },
+    { value: 'appStateOverhead', label: 'appStateOverhead', color: COLORS[11] },
+    { value: 'locationOverhead', label: 'locationOverhead', color: COLORS[12] },
+    { value: 'thermalStateOverhead', label: 'thermalStateOverhead', color: COLORS[13] }
 ]
 const iosMemoryOptions = [
-    { value: 'memoryMB', label: 'memoryMB', color: '#007FFF' }
+    { value: 'memoryMB', label: 'memoryMB', color: COLORS[0] }
 ]
 
 export default class PerfTestDashboard extends React.Component {
     state = {
         perfTestResult: this.props.perfTestResult,
+        testTask: this.props.testTask,
+        perfHistoryList: [],
         androidMemoryInfo: undefined,
         androidBatteryInfo: undefined,
         windowsMemoryInfo: undefined,
         iosEnergyInfo: undefined,
         iosMemoryInfo: undefined,
-        selectedAndroidBatteryOptions: androidBatteryOptions.slice(0, 4),
+        selectedAndroidBatteryOptions: androidBatteryOptions.slice(0, 7),
         selectedAndroidMemoryOptions: androidMemoryOptions.slice(0, 10),
         selectedWindowsMemoryOptions: windowsMemoryOptions.slice(0, 7),
         selectedIosEnergyOptions: iosEnergyOptions.slice(0, 7)
     };
 
+    buildPerfHistoryOptions(historyList) {
+        if (historyList && historyList.length > 0) {
+            let historyData = historyList[0];
```

Comment:
Why is only the 0th element used here and what is the structure of the historyList?

Reply:
They are unused codes, I've removed them. Thanks for reminding.